pkg/agent/openflow/pod_connectivity.go

```diff
@@ -157,16 +163,105 @@ func (f *featurePodConnectivity) initFlows() []binding.Flow {
 	}
 
 	return flows
 }
+
+// trafficControlMarkFlows generates the flows to mark the packets that need to be redirected or mirrored with the value of
+// the destination output port.
+func (f *featurePodConnectivity) trafficControlMarkFlows(srcOfPorts []uint32, dstOfPort uint32, direction v1alpha2.Direction) []binding.Flow {
+	cookieID := f.cookieAllocator.Request(f.category).Raw()
+	var flows []binding.Flow
+	for _, port := range srcOfPorts {
+		if direction == v1alpha2.DirectionIn || direction == v1alpha2.DirectionBoth {
+			flows = append(flows, IngressTrafficControlMarkTable.ofTable.BuildFlow(priorityNormal).
+				Cookie(cookieID).
+				MatchRegFieldWithValue(TargetOFPortField, port).
+				Action().LoadToRegField(TrafficControlOFPortField, dstOfPort).
+				Action().NextTable().
+				Done())
+		}
+		if direction == v1alpha2.DirectionOut || direction == v1alpha2.DirectionBoth {
+			flows = append(flows, EgressTrafficControlMarkTable.ofTable.BuildFlow(priorityNormal).
+				Cookie(cookieID).
+				MatchInPort(port).
+				Action().LoadToRegField(TrafficControlOFPortField, dstOfPort).
+				Action().NextTable().
+				Done())
+		}
+	}
+	return flows
+}
+
+// trafficControlInitFlows generates the initial flows for the traffic control mark value.
+func (f *featurePodConnectivity) trafficControlInitFlows(ofInPort, ofOutPort uint32, action v1alpha2.Action) []binding.Flow {
+	cookieID := f.cookieAllocator.Request(f.category).Raw()
+	var flows []binding.Flow
+	if action == v1alpha2.ActionMirror {
+		// This generates the flow to output packets to the original out port as well as mirror the packets to the ofInPort
+		// which is used to receive mirrored packets in L2ForwardingOutTable.
+		flows = append(flows, L2ForwardingOutTable.ofTable.BuildFlow(priorityHigh+1).
+			Cookie(cookieID).
+			MatchRegFieldWithValue(TrafficControlOFPortField, ofInPort).
+			MatchRegMark(OFPortFoundRegMark).
+			Action().OutputToRegField(TargetOFPortField).
+			Action().Output(ofInPort).
+			Done())
+	} else if action == v1alpha2.ActionRedirect {
+		flows = append(flows,
+			// This generates the flow to forward the redirected packets sourced from the ofOutPort to stageRouting directly
+			// in ClassifierTable. Note that, for the packets which are to be originally output to a tunnel, value of
+			// NXM_NX_TUN_IPV4_DST for the redirected packets needs to be loaded in stageRouting.
+			ClassifierTable.ofTable.BuildFlow(priorityNormal).
+				Cookie(cookieID).
+				MatchInPort(ofOutPort).
+				Action().LoadRegMark(FromRedirectRegMark).
+				Action().GotoStage(stageRouting).
+				Done(),
+			// This generates the flow to forward the packets to be redirected to the ofInPort which is used to receive
+			// the packets.
+			L2ForwardingOutTable.ofTable.BuildFlow(priorityHigh+1).
+				Cookie(cookieID).
+				MatchRegFieldWithValue(TrafficControlOFPortField, ofInPort).
+				MatchRegMark(OFPortFoundRegMark).
+				Action().Output(ofInPort).
```

Comment:
Can we make the flow more generic, like matching a reg which indicates it should be redirect or mirror and `OutputToRegField(TrafficControlOFPortField)`, or `OutputToRegField(TargetOFPortField).OutputToRegField(TrafficControlOFPortField)`?